pkg/antctl/raw/traceflow/command_test.go

```diff
@@ -15,15 +15,91 @@
 package traceflow
 
 import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
 	"testing"
+	"time"
 
+	"github.com/spf13/cobra"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
 
 	"antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	antrea "antrea.io/antrea/pkg/client/clientset/versioned"
+	antreafakeclient "antrea.io/antrea/pkg/client/clientset/versioned/fake"
+)
+
+const (
+	srcPod = "default/Pod-1"
+	dstPod = "default/Pod-2"
+	IP     = "1.1.1.1"
+)
+
+var (
+	pod1 = v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "Pod-1",
+			Namespace: "default",
+		},
+		Spec: v1.PodSpec{
+			NodeName: "Node-1",
+			Containers: []v1.Container{
+				{
+					Ports: []v1.ContainerPort{},
+				},
+			},
+		},
+		Status: v1.PodStatus{
+			NominatedNodeName: "Node-1",
+			PodIP:             "100.96.1.2",
+		},
+	}
+
+	pod2 = v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "Pod-2",
+			Namespace: "default",
+		},
+		Spec: v1.PodSpec{
+			NodeName: "Node-1",
+			Containers: []v1.Container{
+				{
+					Ports: []v1.ContainerPort{},
+				},
+			},
+		},
+		Status: v1.PodStatus{
+			NominatedNodeName: "Node-1",
+			PodIP:             "100.96.1.3",
+		},
+	}
 )
 
 var protocolTCP = int32(6)
 
+func newCommandAndOptions(src, dst, outputType string, timeout time.Duration, liveTraffic, droppedOnly, nowait bool) (*cobra.Command, *opt) {
+	Command := &cobra.Command{
+		RunE: runE,
+		Args: cobra.NoArgs,
+	}
+	var localOption opt
+	Command.Flags().StringVarP(&localOption.source, "source", "S", src, "source of the Traceflow: Namespace/Pod, Pod, or IP")
+	Command.Flags().StringVarP(&localOption.destination, "destination", "D", dst, "destination of the Traceflow: Namespace/Pod, Pod, Namespace/Service, Service or IP")
+	Command.Flags().StringVarP(&localOption.outputType, "output", "o", outputType, "output type: yaml (default), json")
+	Command.Flags().StringVarP(&localOption.flow, "flow", "f", "", "specify the flow (packet headers) of the Traceflow packet, including tcp_src, tcp_dst, tcp_flags, udp_src, udp_dst, ipv6")
+	Command.Flags().BoolVarP(&localOption.liveTraffic, "live-traffic", "L", liveTraffic, "if set, the Traceflow will trace the first packet of the matched live traffic flow")
+	Command.Flags().BoolVarP(&localOption.droppedOnly, "dropped-only", "", droppedOnly, "if set, capture only the dropped packet in a live-traffic Traceflow")
+	Command.Flags().Duration("timeout", timeout, "if set, command returns without retrieving results")
+	Command.Flags().BoolVarP(&localOption.nowait, "nowait", "", nowait, "if set, command returns without retrieving results")
+	return Command, &localOption
+}
```

Comment:
why repeating the code? If it's to avoid side effect to the global Command variable, maybe you could call its resets functions after each test?

Reply:
It is needed because i require different values of option at various places. So, this helps to create custom options.

Reply:
I think you should call `Command.Flags().Set` to set the value, instead of creating a new command with repeated code.